src/Handler/LogDnaHandler.php

```diff
@@ -78,9 +72,20 @@ public function write(array $record): void
     {
         $body = $record['formatted'];
 
-        if (mb_strlen($body, '8bit') > static::LOGDNA_BYTE_LIMIT) {
-            $decodedBody = json_decode($body, true);
-
+        /**
+         * We need to pretty print the metadata JSON before we check its size, as LogDNA's 32KB limit applies to the
+         * number bytes of metadata JSON **AFTER** LogDNA's API (Restify) has parsed and pretty formatted it.
+         *
+         * It **DOES NOT* apply to the number of bytes actually sent in the API request as you'd expect.
+         *
+         * Essentially this is guess work on our part - we're hoping that the size of the JSON pretty printed is roughly
+         * the same as the parsing process on LogDNA's end. It's the best we can do under the circumstances.
+         *
+         * Confirmed the behaviour in ticket 10474.
+         */
+        $decodedBody = json_decode($body, true);
+
+        if (mb_strlen(json_encode($decodedBody['lines'][0]['meta'], JSON_PRETTY_PRINT), '8bit') > static::LOGDNA_META_DATA_LIMIT) {
```

Comment:
We were checking against the whole `$body` before, but now checking against `$decodedBody['lines'][0]['meta']`. Is that right?